test/com/amazon/ion/impl/bin/IonEncoder_1_1Test.java

```diff
@@ -528,4 +651,50 @@ protected Decimal convert(String source) throws ArgumentConversionException {
             return Decimal.valueOf(source);
         }
     }
+
+    /**
+     * Converts a Hex String to a Byte Array for a @Parameterized test
+     */
+    static class HexStringToByteArray extends TypedArgumentConverter<String, byte[]> {
+        protected HexStringToByteArray() {
+            super(String.class, byte[].class);
+        }
+
+        @Override
+        protected byte[] convert(String source) throws ArgumentConversionException {
+            if (source == null) return null;
+            if (source.trim().isEmpty()) return new byte[0];
+            String[] octets = source.split(" ");
+            byte[] result = new byte[octets.length];
+            for (int i = 0; i < octets.length; i++) {
+                result[i] = (byte) Integer.parseInt(octets[i], 16);
+            }
+            return result;
+        }
+    }
+
+    /**
+     * Converts a String of symbol ids to a long[] for a @Parameterized test
+     */
+    static class SymbolIdsToLongArray extends TypedArgumentConverter<String, long[]> {
+        protected SymbolIdsToLongArray() {
+            super(String.class, long[].class);
+        }
+
+        @Override
+        protected long[] convert(String source) throws ArgumentConversionException {
+            if (source == null) return null;
+            List<Long> temp = new ArrayList<>();
+            String[] sids = source.split("\\$");
```

Comment:
Doesn't really matter, but could you replace (with nothing) the question marks, split on spaces, and parse the resulting longs without needing the temp list?

Reply:
The reason for the temp list is because I don't know the size of the array ahead of time. I could eliminate the `temp` list it by counting the `$` before splitting.